toolkit/tools/internal/pkggraph/pkggraph_test.go

```diff
@@ -600,11 +620,20 @@ func TestConditionalLookupBasic(t *testing.T) {
 	assert.NotNil(t, lu)
 	assert.True(t, lu.RunNode.Equal(pkgCRun))
 
-	// Best is D2, which is ver: <=2
+	// Best is D2, which is ver: <=2, BUT since we can't predict the future its not safe to use it
```

Comment:
What is wrong with picking `D2` in this case? I didn't understand the bit about predicting the future, I think.